### PR TITLE
Cleanup Reconciler Tests in order to pick up Genreconciler updates

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

This PR has two commits and is needed for #635 :

### Merge service and deployment tests into TestReconcile

We had separate tests for reconcileService and reconcileDeployment. These tests
were hard to port with the genreconciler update(#635) which changes the Reconciler
interface. Instead of testing these unexported functions, I merged the test
cases for these into the existing TestReconcile function

### Make reconciler tests more readable …

The reconciler tests used variables of the form service1, deployment2 etc.
making it hard to figure out what those objects actually did. This change does
the following:

1. Adds more descriptive names e.g. `eventListenerWithNodeSelector` instead of
`eventlistener6`

2. Adds helper functions `makeEL`, `makeDeployment`,
`makeService` to create the above objects instead of deep-copying and changing
values

3. Fixes a bug I found in the process: The tests were setting the
generatedResourceName in the status of the pre-reconciled resource. Instead it
should be set by the Reconciler. Without this, we weren't actually testing if
the status contained the right `generatedResourceName`

/kind cleanup
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```
NONE
```

### CODE REVIEW

File: pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go [hunk 1]
```diff
@@ -138,486 +123,39 @@ func getEventListenerTestAssets(t *testing.T, r test.Resources) (test.Assets, co
...
+				*ownerRefs,
 			},
-			endResources: test.Resources{
-				Namespaces:     []*corev1.Namespace{namespaceResource},
-				EventListeners: []*v1alpha1.EventListener{eventListener2},
-				Deployments:    []*appsv1.Deployment{deployment2},
+			Labels: map[string]string{
+				"app.kubernetes.io/managed-by": "EventListener",
+				"app.kubernetes.io/part-of":    "Triggers",
+				"eventlistener":                eventListenerName,
```
Comment (savitaashture) [inline pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go:156]:
```suggestion
			Labels: generatedLabels,
```

File: pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go [hunk 2]
```diff
@@ -699,44 +235,153 @@ func TestReconcile(t *testing.T) {
...
+			Name:      generatedResourceName,
+			Namespace: namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*ownerRefs,
+			},
+			Labels: map[string]string{
+				"app.kubernetes.io/managed-by": "EventListener",
+				"app.kubernetes.io/part-of":    "Triggers",
+				"eventlistener":                eventListenerName,
+			},
```
Comment (savitaashture) [inline pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go:259]:
```suggestion
			Labels: generatedLabels,
```

Comment (dibyom) [reply pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go]: done!

File: pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go (hunk 1 above)
Comment (dibyom) [reply pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go]: done!

**Comment by dibyom** [comment]
> Bleh - This is still a bit hard to follow, since the transformation from start to end state are still somewhat obscured since they are not localized with the tests themselves.

Yeap, next PR is to fix that 😄
